src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -290,6 +290,42 @@ protected function stream(string $view, array $parameters = [], StreamedResponse
         return $response;
     }
 
+    /**
+     * Handles a form:
+     *
+     * * if the form is not submitted, the Twig template passed in $view is rendered and a 200 HTTP status code is set
+     * * if the form is submitted but invalid, the Twig template passed in $view is rendered and 422 HTTP status code is set
+     * * if the form is submitted and valid, the entity is saved (only if it is managed by Doctrine ORM), a 306 HTTP status code is set and the Location HTTP header is set to the value of $redirectUrl
+     */
+    public function handleForm(Request $request, FormInterface $form, string $view, ?string $redirectUrl = null, ?callable $onSuccess = null, array $viewParameters = [], ?Response $response = null): Response
+    {
+        $form->handleRequest($request);
+
+        $submitted = $form->isSubmitted();
+        $valid = $submitted && $form->isValid();
+
+        if ($valid) {
+            if (null !== $onSuccess) {
+                return $onSuccess();
+            }
+
+            $data = $form->getData();
+            if ($manager = $this->getDoctrine()->getManagerForClass(get_class($data))) {
```

Comment:
As we already provide glue code with Doctrine in this class and in the form component, and because it's the typical use case, I think that it's nice and expected (but this behavior can be changed with `onSuccess`).